### PR TITLE
[3/5] FWI-3110 - do not fail on empty optional parameters

### PR DESCRIPTION
**Pull request by vitorvezani**
This PR fixes internal FWI-3110

## Checklist
* [X] I have signed the CLA
* [X] I have updated/added any relevant documentation

### What's the goal of this PR?
1 - Improved ways of fetching master-hash via `{branch}`, `origin/{branch}`, `remotes/origin/{branch}`

2 - Make GIT commands optional for fetching the following info:
- `masterHash` 
- `commitMessage`
- `branch`
- `origin`

3 - Add  `CI_RUNNER` env. var support
- so we can show hints on cmd failures
- and be able to send to backend via header "X-CI-Runner"

### What changes did you make?
CI plugin do not fail on some git command errors and show hints

### CODE REVIEW

File: plugins/ci/pkg/ci/git.go [hunk 1]
```diff
@@ -62,41 +78,31 @@ func getGitInfo(baseRepoPath, repoName, baseBranch string) (*gitInfo, error) {
...
-		repoName = origin
-		if strings.Contains(repoName, "@") { // git@github.com URLs are allowed
-			repoNameSplit := strings.Split(repoName, "@")
-			// Take the substring after the last @ to avoid any tokens in an HTTPS URL
-			repoName = repoNameSplit[len(repoNameSplit)-1]
-		} else if strings.Contains(repoName, "//") {
-			repoNameSplit := strings.Split(repoName, "//")
-			repoName = repoNameSplit[len(repoNameSplit)-1]
-		}
-		// Remove "******.com:" prefix and ".git" suffix to get clean $org/$repo structure
```
Comment (ivanfetch-fw) [inline plugins/ci/pkg/ci/git.go:94]: These comments about what is being removed by splitting on a particular character are very helpful to understand why the code does what it does!

File: plugins/ci/pkg/ci/git_test.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+	r, err = getGitInfo(errorOnOptionalStubExecutor, models.GithubActions, ".", "fairwinds/insights-plugins", "main")
+	assert.NoError(t, err)
+	assert.Equal(t, &gitInfo{origin: "", branch: "", masterHash: "", currentHash: "current-hash", commitMessage: "", repoName: "fairwinds/insights-plugins"}, r)
+
+	r, err = getGitInfo(errorOnRequiredStubExecutor, models.GithubActions, ".", "fairwinds/insights-plugins", "main")
+	assert.Error(t, err)
+	assert.Nil(t, r)
+}
+
```
Comment (ivanfetch-fw) [inline plugins/ci/pkg/ci/git_test.go:39]: I like how you've abstracted the executor function to test `getGitInfo`. I do wish the helper functions for tests felt easier for me to follow, but I can't think of how they could be improved at the moment.

**Review by ivanfetch-fw — APPROVED** [review]

File: plugins/ci/pkg/ci/git_test.go (hunk 2 above)
Comment (vitorvezani) [reply plugins/ci/pkg/ci/git_test.go:39]: Yea, I thought of having the return based on how many times the function has been called but sounded like a very obscure approach

File: plugins/ci/pkg/ci/git.go (hunk 1 above)
Comment (vitorvezani) [reply plugins/ci/pkg/ci/git.go:94]: Indeed - also adding tests, so we may clearly see all the expected inputs and their respected outputs

**Pull request merged**